File: SmartBooksFrontend/src/app/models/supplier.ts
import { Address } from './address';

export interface Supplier {
  id: number;
  name: string;
  currencyId: number;
  firstName: string;
  lastName: string;
  phoneNumber: string;
  email: string;
  idNumber: string;
  defaultAddressId?: number;
  categoryId?: number;
  paymentTermId?: number;
  creditLimit: number;
  creditLimitPeriod: number;
  ledgerAccountId?: number;
  balance: number;
  currencyBalance: number;
  accountNumber: string;
  vatNumber: string;
  pin: string;
  bankId: number;
  bankBranchId: number;
  bankAccountNumber: string;
  isTaxAgency: boolean;
  organisationAddresses: Array<Address>;
}

export interface SuppliersList {
  organisations: Array<Supplier>;
  totalCount: number;
}

File: SmartBooks.Domains/Entities/Supplier.cs
namespace SmartBooks.Domains.Entities
{
    public class Supplier : Organisation
    {
        public int? BankId { get; set; }
        public int? BankBranchId { get; set; }
        public string BankAccountNumber { get; set; }
        public bool IsTaxAgency { get; set; }

        public Bank Bank { get; set; }
        public BankBranch BankBranch { get; set; }
    }
}
